services/autoscaling/src/simcore_service_autoscaling/core/settings.py

```diff
@@ -50,23 +50,9 @@ class EC2InstancesSettings(BaseCustomSettings):
         min_length=1,
         description="Defines the AMI (Amazon Machine Image) ID used to start a new EC2 instance",
     )
-    EC2_INSTANCES_MAX_INSTANCES: int = Field(
-        default=10,
-        description="Defines the maximum number of instances the autoscaling app may create",
-    )
-    EC2_INSTANCES_SECURITY_GROUP_IDS: list[str] = Field(
-        ...,
-        min_items=1,
-        description="A security group acts as a virtual firewall for your EC2 instances to control incoming and outgoing traffic"
-        " (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security-groups.html), "
-        " this is required to start a new EC2 instance",
-    )
-    EC2_INSTANCES_SUBNET_ID: str = Field(
-        ...,
-        min_length=1,
-        description="A subnet is a range of IP addresses in your VPC "
-        " (https://docs.aws.amazon.com/vpc/latest/userguide/configure-subnets.html), "
-        "this is required to start a new EC2 instance",
+    EC2_INSTANCES_CUSTOM_BOOT_SCRIPTS: list[str] = Field(
+        default_factory=list,
+        description="script(s) to run on EC2 instance startup (be careful!), each entry is run one after the other using '&&' operator",
```

Comment:
THOUGHT: why not add a `validator` that produces a warning are simply stops the app startup (i.e. that is the time this is resolved)?

Reply:
what kind of validator? you want me to lint the bash scripts there? like using shellcheck? if yes then it's a bit heavy I think...
if not please precise..